packages/cardpay-sdk/sdk/scheduled-payment-module.ts

```diff
@@ -838,10 +838,12 @@ export default class ScheduledPaymentModule {
     });
   }
 
-  async cancelScheduledPayment(scheduledPaymentId: string, hubRootUrl?: string) {
+  async cancelScheduledPayment(scheduledPaymentId: string, hubRootUrl?: string, authToken?: string) {
     let hubAuth = await getSDK('HubAuth', this.ethersProvider, hubRootUrl, this.signer);
     hubRootUrl = await hubAuth.getHubUrl();
-    let authToken = await hubAuth.authenticate();
+    if (!authToken) {
```

Comment:
Do not prompt to authenticate when already authenticated. It is expected the user is authenticated at this point when this is called.